### ISSUE TITLE
NullPointerException while refactoring a field name in a Java class

### ISSUE THREAD

**Issue by mauromol** [groovy/groovy-eclipse#672]
I was renaming a field name in a Java class, requesting to update its getter method name as well. This caused a NPE by Greclipse in a test method of a test class that is using the refactored class. Here is the stack trace:

```
Bundle: org.eclipse.jdt.groovy.core 3.1.0.xx-201808130148-e48
Message: Groovy-Eclipse Type Inferencing: Error visiting method testPopulateAttestazioneDiAvvenutaTrasmissioneMessageFull in class DefaultFatturaElettronicaMessageFromJpaEntityConverterTest
Exception:
java.lang.NullPointerException: null
    at org.eclipse.jdt.groovy.search.FieldReferenceSearchRequestor.acceptASTNode(FieldReferenceSearchRequestor.java:82)
    at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.handleRequestor(TypeInferencingVisitorWithRequestor.java:1867)
    at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.handleSimpleExpression(TypeInferencingVisitorWithRequestor.java:1856)
    at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitConstantExpression(TypeInferencingVisitorWithRequestor.java:989)
    at org.codehaus.groovy.ast.expr.ConstantExpression.visit(ConstantExpression.java:85)
    at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitBinaryExpression(TypeInferencingVisitorWithRequestor.java:776)
    at org.codehaus.groovy.ast.expr.BinaryExpression.visit(BinaryExpression.java:51)
    at org.codehaus.groovy.ast.CodeVisitorSupport.visitReturnStatement(CodeVisitorSupport.java:127)
    at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitReturnStatement(ClassCodeVisitorSupport.java:283)
    at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitReturnStatement(TypeInferencingVisitorWithRequestor.java:1574)
    at org.codehaus.groovy.ast.stmt.ReturnStatement.visit(ReturnStatement.java:49)
    at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:89)
    at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:238)
    at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitBlockStatement(TypeInferencingVisitorWithRequestor.java:848)
    at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:71)
    at org.codehaus.groovy.ast.CodeVisitorSupport.visitClosureExpression(CodeVisitorSupport.java:228)
    at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitClosureExpression(TypeInferencingVisitorWithRequestor.java:970)
    at org.codehaus.groovy.ast.expr.ClosureExpression.visit(ClosureExpression.java:49)
    at org.codehaus.groovy.ast.CodeVisitorSupport.visitListOfExpressions(CodeVisitorSupport.java:328)
    at org.codehaus.groovy.ast.CodeVisitorSupport.visitTupleExpression(CodeVisitorSupport.java:232)
    at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitTupleExpression(TypeInferencingVisitorWithRequestor.java:1698)
    at org.codehaus.groovy.ast.CodeVisitorSupport.visitArgumentlistExpression(CodeVisitorSupport.java:339)
    at org.codehaus.groovy.ast.expr.ArgumentListExpression.visit(ArgumentListExpression.java:76)
    at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitMethodCallExpression(TypeInferencingVisitorWithRequestor.java:1420)
    at org.codehaus.groovy.ast.expr.MethodCallExpression.visit(MethodCallExpression.java:70)
    at org.codehaus.groovy.ast.CodeVisitorSupport.visitExpressionStatement(CodeVisitorSupport.java:123)
    at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitExpressionStatement(ClassCodeVisitorSupport.java:268)
    at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:42)
    at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:89)
    at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:238)
    at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitBlockStatement(TypeInferencingVisitorWithRequestor.java:848)
    at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:71)
    at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClassCodeContainer(ClassCodeVisitorSupport.java:145)
    at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitConstructorOrMethod(TypeInferencingVisitorWithRequestor.java:1081)
    at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitJDT(TypeInferencingVisitorWithRequestor.java:408)
    at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitJDT(TypeInferencingVisitorWithRequestor.java:282)
    at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitCompilationUnit(TypeInferencingVisitorWithRequestor.java:244)
    at org.codehaus.jdt.groovy.integration.internal.GroovyLanguageSupport.maybePerformDelegatedSearch(GroovyLanguageSupport.java:293)
    at org.codehaus.jdt.groovy.integration.LanguageSupportFactory.maybePerformDelegatedSearch(LanguageSupportFactory.java:118)
    at org.eclipse.jdt.internal.core.search.matching.MatchLocator.locateMatches(MatchLocator.java:1275)
    at org.eclipse.jdt.internal.core.search.matching.MatchLocator.locateMatches(MatchLocator.java:1397)
    at org.eclipse.jdt.internal.core.search.matching.MatchLocator.locateMatches(MatchLocator.java:1539)
    at org.eclipse.jdt.internal.core.search.JavaSearchParticipant.locateMatches(JavaSearchParticipant.java:109)
    at org.eclipse.jdt.internal.core.search.BasicSearchEngine.findMatches(BasicSearchEngine.java:247)
    at org.eclipse.jdt.internal.core.search.BasicSearchEngine.search(BasicSearchEngine.java:592)
    at org.eclipse.jdt.core.search.SearchEngine.search(SearchEngine.java:674)
    at org.eclipse.jdt.internal.corext.refactoring.RefactoringSearchEngine.internalSearch(RefactoringSearchEngine.java:143)
    at org.eclipse.jdt.internal.corext.refactoring.RefactoringSearchEngine.search(RefactoringSearchEngine.java:130)
    at org.eclipse.jdt.internal.corext.refactoring.rename.RenameFieldProcessor.getReferences(RenameFieldProcessor.java:608)
    at org.eclipse.jdt.internal.corext.refactoring.rename.RenameFieldProcessor.doCheckFinalConditions(RenameFieldProcessor.java:462)
    at org.eclipse.jdt.internal.corext.refactoring.rename.JavaRenameProcessor.checkFinalConditions(JavaRenameProcessor.java:48)
    at org.eclipse.ltk.core.refactoring.participants.ProcessorBasedRefactoring.checkFinalConditions(ProcessorBasedRefactoring.java:222)
    at org.eclipse.ltk.core.refactoring.CheckConditionsOperation.run(CheckConditionsOperation.java:83)
    at org.eclipse.ltk.core.refactoring.CreateChangeOperation.run(CreateChangeOperation.java:119)
    at org.eclipse.ltk.core.refactoring.PerformChangeOperation.run(PerformChangeOperation.java:207)
    at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2289)
    at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2316)
    at org.eclipse.ltk.internal.ui.refactoring.WorkbenchRunnableAdapter.run(WorkbenchRunnableAdapter.java:86)
    at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
```

This is the body of the mentioned test method:

```groovy
	@Test
	public final void testPopulateAttestazioneDiAvvenutaTrasmissioneMessageFull() {
		def entity = new AttestazioneDiAvvenutaTrasmissioneEntity()
		entity.with {
			identificativoSdi = '123'
			nomeFile = 'nomeFile'
			dataOraRicezione = Date.parse('yyyy-MM-dd HH:mm:ss', '2014-10-17 15:46:01')
			riferimentoArchivio = new RiferimentoArchivioEmbeddable()
			codiceDestinatario = 'codiceDestinatario'
			descrizioneDestinatario = 'descrizioneDestinatario'
			messageId = 'messageId'
			pecMessageId = 'pecMessageId'
			note = 'note'
			hashFileOriginale = 'hashFileOriginale'
		}
		def model = new AttestazioneDiAvvenutaTrasmissioneMessage()
		assertSame(model, c.populateAttestazioneDiAvvenutaTrasmissioneMessage(entity, model))
		assertEquals('123', model.identificativoSdi)
		assertEquals('nomeFile', model.nomeFile)
		assertEquals('2014-10-17 15:46:01', model.dataOraRicezione.format('yyyy-MM-dd HH:mm:ss'))
		assertNotNull(model.riferimentoArchivio)
		assertEquals('codiceDestinatario', model.codiceDestinatario)
		assertEquals('descrizioneDestinatario', model.descrizioneDestinatario)
		assertEquals('messageId', model.messageId)
		assertEquals('pecMessageId', model.pecMessageId)
		assertEquals('note', model.note)
		assertEquals('hashFileOriginale', model.hashFileOriginale)
	}
```

Please note, the reference that should have been updated is on the last line: it was `model.hashFileOriginale` and should have been updated to `model.hashFile` (because I changed the field name from `hashFileOriginale` to `hashFile` and the corresponding getter method from `getHashFileOriginale()` to `getHashFile()`). However, it was changed in a wrong way to: `model.getHashFile` (which is neither a property-style access, nor a getter method access).

Please note that there's just a particular thing: the Java bean is something like that:

```java
public class MyBean() {
  private Wrapper<String> hashFileOriginale = new Wrapper<>();

  public String getHashFileOriginale() { return hashFileOriginale.get(); }
  public void setHashOriginale(String v) { hashFileOrignale.set(v); }
}
```

In other words, the renamed field is not a classic Java Bean field but it's wrapped (and the getter and the setter perform the unwrapping).

I tried to isolate in a small test case but I was unsuccessful. I could get the exact same exception (NPE at `org.eclipse.jdt.groovy.search.FieldReferenceSearchRequestor.acceptASTNode(FieldReferenceSearchRequestor.java:82)`) but on a Groovy file that has nothing to do with the refactoring I'm performing (it contains a "foo" reference but for a field of a totally different class). If I just put the above Java bean and Groovy classes on the classpath, I don't get the exception.

Hope this information is enough to drive you to the right direction, otherwise let me know.

**Comment by mauromol** [comment]
If I use this small test case, I don't get the NPE, but the refactoring on the Groovy class is not performed:

```java
package test29;

import java.util.concurrent.atomic.AtomicInteger;

public class TestRefactor {

	private AtomicInteger foo;

	
	public int getFoo() {
		return this.foo.get();
	}	

	public void setFoo(int foo) {
		this.foo.set(foo);
	}
}
```

```groovy
package test29

import groovy.transform.CompileDynamic
import groovy.transform.CompileStatic

@CompileStatic
class Test29 {

	void bar() {
		def t = new TestRefactor()
		def f = t.foo
	}
}
```

Now, select `foo` property in TestRefactor class and hit Alt+Shift+R twice, so the "Rename Field" dialog appears: change `foo` to `foobar` and check "Rename getter: ...". Hit ok: `def f = t.foo` in `Test29` is not changed to `def f = t.foobar`.

**Comment by mauromol** [comment]
Ok, I was able to reproduce the NPE but putting these files on the classpath:

```groovy
package test29;

import java.util.concurrent.atomic.AtomicInteger;

public class TestRefactor {

	private AtomicInteger foo;

	
	public int getFoo() {
		return this.foo.get();
	}	

	public void setFoo(int foo) {
		this.foo.set(foo);
	}
}
```

```java
package test29

import groovy.transform.CompileStatic

@CompileStatic
class Test29 {

	void bar() {
		def t = new TestRefactor()
		def f = t.foo
	}
}
```

```java
package test22;
public class MyBean {
	private String foo;
	public void setFoo(String foo) {
		this.foo = foo;
	}
	public String getFoo() {
		return foo;
	}
}
```

```groovy
package test22

import groovy.transform.CompileStatic

@CompileStatic
class Test22 {
	public Test22() {
		def b = new MyBean(foo: 'c')
		def m = b.&setFoo
		m('foo')
		def c = b.foo
	}
}
```

Now select `TestRefactor.foo`, hit Alt+Shift+R twice and change it to `foobar`, including getter. I get:

```
java.lang.NullPointerException
	at org.eclipse.jdt.groovy.search.FieldReferenceSearchRequestor.acceptASTNode(FieldReferenceSearchRequestor.java:82)
	at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.handleRequestor(TypeInferencingVisitorWithRequestor.java:1867)
	at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.handleSimpleExpression(TypeInferencingVisitorWithRequestor.java:1856)
	at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitConstantExpression(TypeInferencingVisitorWithRequestor.java:989)
	at org.codehaus.groovy.ast.expr.ConstantExpression.visit(ConstantExpression.java:85)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitListOfExpressions(CodeVisitorSupport.java:328)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitTupleExpression(CodeVisitorSupport.java:232)
	at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitTupleExpression(TypeInferencingVisitorWithRequestor.java:1698)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitArgumentlistExpression(CodeVisitorSupport.java:339)
	at org.codehaus.groovy.ast.expr.ArgumentListExpression.visit(ArgumentListExpression.java:76)
	at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitMethodCallExpression(TypeInferencingVisitorWithRequestor.java:1420)
	at org.codehaus.groovy.ast.expr.MethodCallExpression.visit(MethodCallExpression.java:70)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitExpressionStatement(CodeVisitorSupport.java:123)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitExpressionStatement(ClassCodeVisitorSupport.java:268)
	at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:42)
	at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitConstructorOrMethod(TypeInferencingVisitorWithRequestor.java:1085)
	at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitJDT(TypeInferencingVisitorWithRequestor.java:408)
	at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitJDT(TypeInferencingVisitorWithRequestor.java:282)
	at org.eclipse.jdt.groovy.search.TypeInferencingVisitorWithRequestor.visitCompilationUnit(TypeInferencingVisitorWithRequestor.java:244)
	at org.codehaus.jdt.groovy.integration.internal.GroovyLanguageSupport.maybePerformDelegatedSearch(GroovyLanguageSupport.java:293)
	at org.codehaus.jdt.groovy.integration.LanguageSupportFactory.maybePerformDelegatedSearch(LanguageSupportFactory.java:118)
	at org.eclipse.jdt.internal.core.search.matching.MatchLocator.locateMatches(MatchLocator.java:1267)
	at org.eclipse.jdt.internal.core.search.matching.MatchLocator.locateMatches(MatchLocator.java:1389)
	at org.eclipse.jdt.internal.core.search.matching.MatchLocator.locateMatches(MatchLocator.java:1521)
	at org.eclipse.jdt.internal.core.search.JavaSearchParticipant.locateMatches(JavaSearchParticipant.java:122)
	at org.eclipse.jdt.internal.core.search.BasicSearchEngine.findMatches(BasicSearchEngine.java:236)
	at org.eclipse.jdt.internal.core.search.BasicSearchEngine.search(BasicSearchEngine.java:583)
	at org.eclipse.jdt.core.search.SearchEngine.search(SearchEngine.java:608)
	at org.eclipse.jdt.internal.corext.refactoring.RefactoringSearchEngine.internalSearch(RefactoringSearchEngine.java:143)
	at org.eclipse.jdt.internal.corext.refactoring.RefactoringSearchEngine.search(RefactoringSearchEngine.java:130)
	at org.eclipse.jdt.internal.corext.refactoring.rename.RenameFieldProcessor.getReferences(RenameFieldProcessor.java:608)
	at org.eclipse.jdt.internal.corext.refactoring.rename.RenameFieldProcessor.doCheckFinalConditions(RenameFieldProcessor.java:462)
	at org.eclipse.jdt.internal.corext.refactoring.rename.JavaRenameProcessor.checkFinalConditions(JavaRenameProcessor.java:48)
	at org.eclipse.ltk.core.refactoring.participants.ProcessorBasedRefactoring.checkFinalConditions(ProcessorBasedRefactoring.java:222)
	at org.eclipse.ltk.core.refactoring.CheckConditionsOperation.run(CheckConditionsOperation.java:83)
	at org.eclipse.ltk.core.refactoring.CreateChangeOperation.run(CreateChangeOperation.java:119)
	at org.eclipse.ltk.core.refactoring.PerformChangeOperation.run(PerformChangeOperation.java:207)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2240)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2267)
	at org.eclipse.ltk.internal.ui.refactoring.WorkbenchRunnableAdapter.run(WorkbenchRunnableAdapter.java:86)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
```

The strange thing is that `def f = t.foo` in `Test29` is now changed to `def f = t.foobar` even if `@CompileStatic` is used!! :-O

The last piece of the puzzle that I can't reproduce in a small test case is the totally wrong refactoring result (to `model.getHashFile`) that was produced in my real project...

**Comment by eric-milles** [comment]
Ready to test.  I'll look into the renaming issue separately.

**Comment by mauromol** [comment]
I'm testing with 3.1.0.xx-201808132239-e47. Unfortunately, now whichever test case I try, I get the following after clicking "OK" in the Rename Field dialog:

```
eclipse.buildId=4.7.3.M20180330-0640
java.version=1.8.0_181
java.vendor=Oracle Corporation
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=it_IT
Command-line arguments:  -os win32 -ws win32 -arch x86_64 -data file:/D:/ws/47/DCS-FATTURAPA-trunk/

org.eclipse.ltk.ui.refactoring
Error
Tue Aug 14 10:11:50 CEST 2018
Internal Error

java.lang.reflect.InvocationTargetException
	at org.eclipse.jface.operation.ModalContext.run(ModalContext.java:398)
	at org.eclipse.ltk.internal.ui.refactoring.RefactoringWizardDialog2.run(RefactoringWizardDialog2.java:321)
	at org.eclipse.ltk.ui.refactoring.RefactoringWizard.internalPerformFinish(RefactoringWizard.java:636)
	at org.eclipse.ltk.ui.refactoring.UserInputWizardPage.performFinish(UserInputWizardPage.java:145)
	at org.eclipse.ltk.ui.refactoring.RefactoringWizard.performFinish(RefactoringWizard.java:710)
	at org.eclipse.ltk.internal.ui.refactoring.RefactoringWizardDialog2.okPressed(RefactoringWizardDialog2.java:447)
	at org.eclipse.jface.dialogs.Dialog.buttonPressed(Dialog.java:466)
	at org.eclipse.jface.dialogs.Dialog.lambda$0(Dialog.java:619)
	at org.eclipse.swt.events.SelectionListener$1.widgetSelected(SelectionListener.java:81)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:249)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4428)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1079)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4238)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3817)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:818)
	at org.eclipse.jface.window.Window.open(Window.java:794)
	at org.eclipse.ltk.ui.refactoring.RefactoringWizardOpenOperation$1.run(RefactoringWizardOpenOperation.java:188)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:70)
	at org.eclipse.ltk.ui.refactoring.RefactoringWizardOpenOperation.run(RefactoringWizardOpenOperation.java:203)
	at org.eclipse.ltk.ui.refactoring.RefactoringWizardOpenOperation.run(RefactoringWizardOpenOperation.java:122)
	at org.eclipse.jdt.internal.ui.refactoring.actions.RefactoringStarter.activate(RefactoringStarter.java:38)
	at org.eclipse.jdt.internal.ui.refactoring.UserInterfaceStarter.activate(UserInterfaceStarter.java:62)
	at org.eclipse.jdt.internal.ui.refactoring.reorg.RenameUserInterfaceStarter.activate(RenameUserInterfaceStarter.java:32)
	at org.eclipse.jdt.ui.refactoring.RenameSupport.openDialog(RenameSupport.java:160)
	at org.eclipse.jdt.ui.refactoring.RenameSupport.openDialog(RenameSupport.java:116)
	at org.eclipse.jdt.internal.ui.refactoring.reorg.RenameLinkedMode.startFullDialog(RenameLinkedMode.java:468)
	at org.eclipse.jdt.internal.ui.refactoring.actions.RenameJavaElementAction.doRun(RenameJavaElementAction.java:157)
	at org.eclipse.jdt.internal.ui.refactoring.actions.RenameJavaElementAction.run(RenameJavaElementAction.java:148)
	at org.eclipse.jdt.ui.actions.RenameAction.run(RenameAction.java:118)
	at org.eclipse.jdt.ui.actions.SelectionDispatchAction.dispatchRun(SelectionDispatchAction.java:275)
	at org.eclipse.jdt.ui.actions.SelectionDispatchAction.run(SelectionDispatchAction.java:249)
	at org.eclipse.jface.action.Action.runWithEvent(Action.java:473)
	at org.eclipse.jface.commands.ActionHandler.execute(ActionHandler.java:118)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:92)
	at sun.reflect.GeneratedMethodAccessor72.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:55)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:305)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:239)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:132)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:152)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:494)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:487)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:210)
	at org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.executeCommand(KeyBindingDispatcher.java:287)
	at org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.press(KeyBindingDispatcher.java:527)
	at org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.processKeyEvent(KeyBindingDispatcher.java:577)
	at org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.filterKeySequenceBindings(KeyBindingDispatcher.java:385)
	at org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.access$0(KeyBindingDispatcher.java:331)
	at org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher$KeyDownFilter.handleEvent(KeyBindingDispatcher.java:88)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
	at org.eclipse.swt.widgets.Display.filterEvent(Display.java:1271)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1078)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1103)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1088)
	at org.eclipse.swt.widgets.Control.traverse(Control.java:4331)
	at org.eclipse.swt.widgets.Control.translateMnemonic(Control.java:4173)
	at org.eclipse.swt.widgets.Composite.translateMnemonic(Composite.java:1272)
	at org.eclipse.swt.widgets.Control.translateMnemonic(Control.java:4191)
	at org.eclipse.swt.widgets.Display.translateMnemonic(Display.java:4978)
	at org.eclipse.swt.widgets.Display.filterMessage(Display.java:1289)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3813)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1150)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1039)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:680)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:594)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:151)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1499)
Caused by: org.eclipse.core.runtime.AssertionFailedException: assertion failed: 
	at org.eclipse.core.runtime.Assert.isTrue(Assert.java:110)
	at org.eclipse.core.runtime.Assert.isTrue(Assert.java:96)
	at org.eclipse.jdt.internal.corext.refactoring.rename.RenameFieldProcessor.checkAccessorDeclarations(RenameFieldProcessor.java:523)
	at org.eclipse.jdt.internal.corext.refactoring.rename.RenameFieldProcessor.checkAccessor(RenameFieldProcessor.java:501)
	at org.eclipse.jdt.internal.corext.refactoring.rename.RenameFieldProcessor.doCheckFinalConditions(RenameFieldProcessor.java:475)
	at org.eclipse.jdt.internal.corext.refactoring.rename.JavaRenameProcessor.checkFinalConditions(JavaRenameProcessor.java:48)
	at org.eclipse.ltk.core.refactoring.participants.ProcessorBasedRefactoring.checkFinalConditions(ProcessorBasedRefactoring.java:222)
	at org.eclipse.ltk.core.refactoring.CheckConditionsOperation.run(CheckConditionsOperation.java:83)
	at org.eclipse.ltk.core.refactoring.CreateChangeOperation.run(CreateChangeOperation.java:119)
	at org.eclipse.ltk.core.refactoring.PerformChangeOperation.run(PerformChangeOperation.java:207)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2240)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2267)
	at org.eclipse.ltk.internal.ui.refactoring.WorkbenchRunnableAdapter.run(WorkbenchRunnableAdapter.java:86)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
```

**Comment by mauromol** [comment]
My real project has changed a bit, so I can't test the actual same situation that got me to the NPE, but I performed a very similar refactoring and it works (without the NPE), apart from the fact that if I check "rename getter" but not "rename setter", Greclipse still refactors property-style write accesses with the new name, hence generating a compilation error. But I may open a new issue for this.

Still, I don't understand why I can't perform the refactoring in the other test-case project. 
Please note I'm using Photon on my real project (where it works), while I'm using Oxygen on the test case project (where I get the assertion error): could this be the cause?

**Comment by eric-milles** [comment]
This is the Photon version of RenameFieldProcessor.checkAccessorDeclarations
```java
	private RefactoringStatus checkAccessorDeclarations(IProgressMonitor pm, IMethod existingAccessor) throws CoreException{
		RefactoringStatus result= new RefactoringStatus();
		SearchPattern pattern= SearchPattern.createPattern(existingAccessor, IJavaSearchConstants.DECLARATIONS, SearchUtils.GENERICS_AGNOSTIC_MATCH_RULE);
		IJavaSearchScope scope= SearchEngine.createHierarchyScope(fField.getDeclaringType());
		SearchResultGroup[] groupDeclarations= RefactoringSearchEngine.search(pattern, scope, pm, result);
		Assert.isTrue(groupDeclarations.length > 0);
```

**Comment by eric-milles** [comment]
The property-style access of `getFoo` in:
```groovy
	void bar() {
		def t = new TestRefactor()
		def f = t.foo
	}
```
is identified as a field reference by this block in `FieldReferenceSearchRequestor`.

```java
    public VisitStatus acceptASTNode(ASTNode node, TypeLookupResult result, IJavaElement enclosingElement) {
       ...
        if (node instanceof ConstantExpression) {
            // check for "foo.bar" where "bar" refers to "getBar()" or "setBar(...)" with backing field or property
            if (fieldName.equals(((ConstantExpression) node).getText()) && (result.confidence == TypeConfidence.UNKNOWN ||
                    result.declaringType.getField(fieldName) != null || result.declaringType.getProperty(fieldName) != null)) {
                doCheck = true;
                isAssignment = EqualityVisitor.checkForAssignment(node, result.enclosingAssignment);
                start = node.getStart();
                end = node.getEnd();
            }
        }
```

This needs to be seen as a method reference for the renaming to work properly.

**Comment by mauromol** [comment]
Hi Eric, I'm a bit lost here, I still can't perform the rename refactoring in Oxygen, it was working properly (apart from the NPE reported here) in 3.1.0.xx-201808130148-e48, while now it fails even before because of the aforementioned AssertionFailedException.

**Comment by eric-milles** [comment]
There is a build running now that cleans up reference searching.  Can you update when it's complete and try your assertion fail scenario again?  If the error is still present, can you open a new issue with steps specific to recreating the assertion failure?

**Comment by mauromol** [comment]
Ok, tomorrow I will test the new version and let you know, thank you!

**Comment by mauromol** [comment]
Weird, after cleaning the project, I can again perform the refactoring even in Oxygen... Probably there was something corrupted.
Now, I easily see the renaming problem described here and specifically in #678.
